gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -857,6 +914,13 @@ protected void afterExecute(Runnable r, Throwable t) {
 
           // Compaction job of a dataset is successful.
           jobRunner.getDataset().setState(COMPACTION_COMPLETE);
+
+          // Add a new MPCompactor job if recompaction is necessary.
+          if (jobRunner.getDataset().needToRecompact()) {
+            for (Dataset recompactDataset: MRCompactor.this.createDatasetForRecompact(jobRunner.getDataset())) {
+              MRCompactor.this.runCompactionForDataset(recompactDataset, true);
```

Comment:
You just need to set the dataset state to `VERIFIED` here. The recompaction job will then be submitted in `submitCompactionJobsAndWaitForCompletion()`.